huaweicloud/services/cfw/resource_huaweicloud_cfw_lts_log.go

```diff
@@ -0,0 +1,332 @@
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jmespath/go-jmespath"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+// @API CFW POST /v1/{project_id}/cfw/logs/configuration
+// @API CFW GET /v1/{project_id}/cfw/logs/configuration
+// @API CFW PUT /v1/{project_id}/cfw/logs/configuration
+func ResourceLtsLog() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceLtsLogCreate,
+		ReadContext:   resourceLtsLogRead,
+		UpdateContext: resourceLtsLogUpdate,
+		DeleteContext: resourceLtsLogDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"fw_instance_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `The ID of the firewall.`,
+			},
+			"lts_log_group_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `LTS log group ID.`,
+			},
+			"lts_attack_log_stream_enable": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: `LTS attack log stream switch.`,
+			},
+			"lts_access_log_stream_enable": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: `LTS access log stream switch.`,
+			},
+			"lts_flow_log_stream_enable": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: `LTS flow log stream switch.`,
+			},
+			"lts_attack_log_stream_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `LTS attack log stream ID.`,
+			},
+			"lts_access_log_stream_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `LTS access log stream ID.`,
+			},
+			"lts_flow_log_stream_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `LTS flow log stream ID.`,
+			},
+		},
+	}
+}
+
+func resourceLtsLogCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var (
+		httpUrl = "v1/{project_id}/cfw/logs/configuration"
+		product = "cfw"
+	)
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating CFW Client: %s", err)
+	}
+
+	path := client.Endpoint + httpUrl
+	path = strings.ReplaceAll(path, "{project_id}", client.ProjectID)
+	path += fmt.Sprintf("?fw_instance_id=%s", d.Get("fw_instance_id"))
+
+	opt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
```

Comment:
it is better to remove the default ok codes, and fix other places

Reply:
fixed